### ISSUE TITLE
[Feature Request] @OnDissociate currently only takes effect on bidirectional foreign key associations. I hope it can be changed to take effect as long as there is only an one-way association

### ISSUE THREAD

**Issue by ClearPlume** [babyfish-ct/jimmer#89]
As mentioned, `@OnDissociate` currently only works when two entities have a bidirectional association, such as the following, a one-to-one association is the same. 

ScanDimension
```kotlin
@Entity
interface ScanDimension {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Int

    @ManyToOne
    @OnDissociate(DissociateAction.DELETE)
    val group: ScanDimensionGroup
}
```

ScanDimensionGroup
```kotlin
@Entity
interface ScanDimensionGroup {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Int

    @OneToMany(mappedBy = "group")
    val dimensions: List<ScanDimension>
}
```

`ScanDimensionGroup#dimensions` is completely useless in the actual logic, it is placed here only to make `@OnDissociate` take effect.

**Comment by babyfish-ct** [comment]
Try 0.7.55(Untagged version)

**Comment by ClearPlume** [comment]
Fixed with 0.7.55
